attelo/decoding/mst.py

```diff
@@ -1,88 +1,116 @@
 '''
 Created on Jun 27, 2012
 
-@author: stergos
-
-
+@author: stergos, jrmyp
 '''
 
-from collections import defaultdict
-from math import log
+from __future__ import print_function
 import sys
 
+from collections import defaultdict
+from scipy.special import logit
+
 from depparse.graph import Digraph
 from .interface import Decoder
 
 # pylint: disable=too-few-public-methods
 
+# Required to work with depparse, working with a "minimum" score of 1e-100
+# Unless we have more than 1e10 nodes, combined scores shouldn't reach it
+MAX_SCORE = 1e90
+MIN_SCORE = -MAX_SCORE
 
-def get_root(_):
-    """ This is used for the construction of the graph. Since we are
-    using the Chu-Liu Edmonds algorithm in it's dependency parsing
-    incarnation, there should be a node that is the root, i.e.  a node
-    that has no incoming edges. This function is supposed to return
-    that node. For the moment it always returns the first
-    node. Syntactic analysis should be sufficient to provide the real
-    head node.
-    """
-    return '1'
+def _get_root(edus):
+    """ Returns the root node for MST/MSDAG algorithm
```

Comment:
I would have kept this part of the comment, personally, but will leave this to your discretion: This is used for the construction of the graph. Since we are using the Chu-Liu Edmonds algorithm in it's dependency parsing incarnation, there should be a node that is the root, i.e.  a node that has no incoming edges